app/main.py

```diff
@@ -1,8 +1,40 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.gender = None
+        self.spouse = None
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def assign_gender(self, gender: str) -> None:
+        self.gender = gender
+
+    def set_spouse(self, spouse_name: str) -> None:
+        if spouse_name in Person.people:
+            self.spouse = Person.people[spouse_name]
+            self.spouse.spouse = self
```

Comment:
you can solve this task without those functions, your func "create_person_list()" can add new atributes to instance with **_name from dictionary.wife = name from dictionary by "wife" in list_**
But i approve u, couse maybe mentor doesn't agree with me 😅